include/opendb/db.h

```diff
@@ -3582,7 +3582,7 @@ class dbVia : public dbObject
     ///
     /// Get the via params used to generate this via.
     ///
-    void getViaParams( dbViaParams & params );
+    void getViaParams( dbViaParams & params_return );
```

Comment:
The renaming is important so that SWIG can distinguish between the getter and setter function because they have different handling, otherwise SWIG will assume that the setViaParams takes no arguments.

Also you cannot switch on a string, it can be done for output typemap since you are switching on enum (can add that), but not the input nor the typecheck.

Reply:
The renaming is important so that SWIG can distinguish between the getter and setter function because they have different handling, otherwise SWIG will assume that the setViaParams takes no arguments.

Also you cannot switch on a string, it can be done for output typemap since you are switching on enum (can add that), but not the input nor the typecheck.

Reply:
Thanks for the clarifications.
I thought that all the values are already enums and we can switch over them. But if this is not the case and it takes time, feel free to ignore this comments.